core/lib/misc.js

```diff
@@ -190,7 +190,7 @@ const completeUserProfile = (user, email, name) => {
     }));
   } catch (err) { throw new Meteor.Error('email-duplicate', 'Email already exists'); }
 
-  Meteor.users.update(user._id, { $unset: { 'profile.guest': true, username: true } });
+  Meteor.users.update(user._id, { $unset: { username: true } });
```

Comment:
You want to keep the "guest" attribute even on users who have created an account? 🤔

Reply:
Yes, as explicated above, we can discuss it in lemverse if you want